### ISSUE TITLE
Test flake: oc tag can fail with imageStream resource conflict

### ISSUE THREAD

**Issue by ironcladlou** [openshift/origin#5526]
Looks like another test flake resulting from concurrent imageStream updates which is slightly different than the ISM fix in #5455.

```
Waiting for the import to complete, CTRL+C to stop waiting.
The import completed successfully.

Error from server: imageStream "newrepo" cannot be updated: the object has been modified; please apply your changes to the latest version and try again
!!! Error in test/cmd/images.sh:110
	'oc tag mysql:5.5 newrepo:latest' exited with status 1
Call stack:
	1: test/cmd/images.sh:110 main(...)
Exiting with status 1
!!! Error in hack/test-cmd.sh:276
	'${test}' exited with status 1
Call stack:
	1: hack/test-cmd.sh:276 main(...)
Exiting with status 1
make: *** [test-int-plus] Error 1
```

@smarterclayton suggested we might need the same sort of idempotence fix in the imageStream REST handler.

**Comment by smarterclayton** [comment]
Candidate, very irritating.  We can cut if necessary.

**Comment by liggitt** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/3902/console

**Comment by liggitt** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/3901/console

**Comment by smarterclayton** [comment]
Looking at

**Comment by smarterclayton** [comment]
Fixed in #5626 but I don't think we necessarily do this for 3.1.  Need to make a call tomorrow.

**Comment by liggitt** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/3919/console
